### PR TITLE
Configure github actions

### PR DESCRIPTION
**Pull request by sgtsquiggs**
travis-ci is outta here

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK 11
+      uses: actions/setup-java@v3
+      with:
+        java-version: '11'
```
Comment (ahmedre) [inline .github/workflows/gradle.yml]: Should we consider using 17 here?

**Review by ahmedre — DISMISSED** [review]
Thanks!

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (sgtsquiggs) [reply .github/workflows/gradle.yml]: we can use a matrix to ensure it works in several versions - which versions should we test against?

Comment (ahmedre) [reply .github/workflows/gradle.yml]: 11 and 17 since they are LTS releases and then the latest (20?)

Comment (sgtsquiggs) [reply .github/workflows/gradle.yml]: test/build fails currently in 17 with:
```General error during semantic analysis: Unsupported class file major version 61```

Comment (sgtsquiggs) [reply .github/workflows/gradle.yml]: gotta be gradle dependencies, right?

Comment (ahmedre) [reply .github/workflows/gradle.yml]: Bump Gradle in `gradle/wrapper/gradle-wrapper.properties` to point to 7.4.2 or higher - working for me locally on Java 17 after that change.

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    - uses: actions/checkout@v3
+    - name: Set up JDK ${{ matrix.version }}
+      uses: actions/setup-java@v3
+      with:
+        java-version: ${{ matrix.version }}
+        distribution: 'temurin'
+    - name: Set up Gradle
+      uses: gradle/gradle-build-action@v2
+    - name: Run test with Gradle Wrapper
+      run: ./gradlew --debug test
```
Comment (ahmedre) [inline .github/workflows/gradle.yml]: let's remove `--debug`?

**Comment by ahmedre** [comment]
thanks, just one minor comment but lgtm otherwise!
will add enhancements to upload the test results in the future.

**Review by ahmedre — APPROVED** [review]

**Comment by ahmedre** [comment]
Thank you!

**Pull request merged**